python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -314,50 +328,143 @@ cdef class Column:
             c_result = make_column_from_scalar(dereference(slr.get()), size)
         return Column.from_libcudf(move(c_result))
 
-    @staticmethod
-    def from_cuda_array_interface_obj(object obj):
-        """Create a Column from an object with a CUDA array interface.
+    @classmethod
+    def from_array_interface(cls, obj):
+        """
+        Create a Column from an object implementing the NumPy Array Interface.
 
         Parameters
         ----------
         obj : object
-            The object with the CUDA array interface to create a column from.
+            Must implement the `__array_interface__` protocol.
+
+        Raises
+        ------
+        NotImplementedError
+            This method is not yet implemented.
+        """
+        raise NotImplementedError(
+            "Converting to a pylibcudf Column is not yet implemented."
+        )
+
+    @classmethod
+    def from_cuda_array_interface(cls, obj):
+        """
+        Create a Column from an object implementing the CUDA Array Interface.
+
+        Parameters
+        ----------
+        obj : object
+            Must implement the `__cuda_array_interface__` protocol.
 
         Returns
         -------
         Column
-            A Column containing the data from the CUDA array interface.
-
-        Notes
-        -----
-        Data is not copied when creating the column. The caller is
-        responsible for ensuring the data is not mutated unexpectedly while the
-        column is in use.
+            A 1D or 2D list column, depending on the input shape.
+
+        Raises
+        ------
+        ValueError
+            If the object is not 1D or 2D, or is not C-contiguous.
+        ImportError
+            If CuPy is not installed.
         """
-        data = gpumemoryview(obj)
-        iface = data.__cuda_array_interface__
-        if iface.get('mask') is not None:
-            raise ValueError("mask not yet supported.")
+        if np is None or cp is None:
```

Comment:
Technically we only need this in the `elif len(shape) == 2:` case?

Reply:
Yes, thanks